lib/libc/minimal/source/time/time.c

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2021 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <posix/time.h>
```

Comment:
If this is included for just clock_gettime prototype, I'd suggest to include a comment on that. Because otherwise it's a bit strange dependency, which can lead to further include dependency hell.

Reply:
Yes, it is for `clock_gettime()` prototype. You mean dependency hell because `posix/time.h` might fetch `time.h` or the other way around? Fortunately, this is just a C source file, not a header file, so I was not expecting any problems with dependencies.

Reply:
Nonetheless, the include should be conditional as it's pulling in something that isn't part of libc.

Reply:
Conditional on what? `CONFIG_POSIX_CLOCK`? Currently whole file is compiled only when `CONFIG_POSIX_CLOCK=y`.

Reply:
> You mean dependency hell because posix/time.h

Because posix/time.h is a full POSIX header which might have/pull things which aren't available in minimal libc or done differently (incompatibly) in it. Put it differently, at any time compiling this source file may break. Some time ago, I spent a lot of effort to straighten up dependencies between minlibc/newlib/posix, and that wasn't even done completely, and this patch is kinda retrograde action in that regard.

Reply:
> Conditional on what? `CONFIG_POSIX_CLOCK`? Currently whole file is compiled only when `CONFIG_POSIX_CLOCK=y`.

OK, then the comment is certainly desirable.

#18892 is also relevant.  It's not clear that this change is a good idea.

Reply:
Understood. I've added a comment, as suggested. Let me know if there is something more that I can do.